frssci/online/R3BOnlineSpectraFrsSciVsMusli.cxx

```diff
@@ -0,0 +1,414 @@
+/******************************************************************************
+ *   Copyright (C) 2019 GSI Helmholtzzentrum für Schwerionenforschung GmbH    *
+ *   Copyright (C) 2019-2024 Members of R3B Collaboration                     *
+ *                                                                            *
+ *             This software is distributed under the terms of the            *
+ *                 GNU General Public Licence (GPL) version 3,                *
+ *                    copied verbatim in the file "LICENSE".                  *
+ *                                                                            *
+ * In applying this license GSI does not waive the privileges and immunities  *
+ * granted to it by virtue of its status as an Intergovernmental Organization *
+ * or submit itself to any jurisdiction.                                      *
+ ******************************************************************************/
+
+#include "FairLogger.h"
+#include "FairRootManager.h"
+#include "FairRunOnline.h"
+#include "FairRuntimeDb.h"
+#include "TCanvas.h"
+#include "TClonesArray.h"
+#include "TFolder.h"
+#include "TH1.h"
+#include "TH2.h"
+#include "THttpServer.h"
+#include "TMath.h"
+
+#include "R3BEventHeader.h"
+#include "R3BFrsSciPosCalData.h"
+#include "R3BFrsSciTofCalData.h"
+#include "R3BMusliCalData.h"
+#include "R3BMusliHitData.h"
+#include "R3BOnlineSpectraFrsSciVsMusli.h"
+
+R3BOnlineSpectraFrsSciVsMusli::R3BOnlineSpectraFrsSciVsMusli()
+    : R3BOnlineSpectraFrsSciVsMusli("FrsSciOnlineSpectra", 1)
+{
+}
+
+R3BOnlineSpectraFrsSciVsMusli::R3BOnlineSpectraFrsSciVsMusli(const char* name, Int_t iVerbose)
+    : FairTask(name, iVerbose)
+    , fFrsSci_PosCal(NULL)
+    , fFrsSci_TofCal(NULL)
+    , fMusli_Cal(NULL)
+    , fMusli_Hit(NULL)
+    , fNEvents(0)
+    , fIdS2(1)
+    , fIdCaveC(3)
+{
+}
+
+R3BOnlineSpectraFrsSciVsMusli::~R3BOnlineSpectraFrsSciVsMusli()
+{
+    LOG(debug) << "R3BOnlineSpectraFrsSciVsMusli::Destructor";
+    if (fFrsSci_PosCal)
+        delete fFrsSci_PosCal;
+    if (fFrsSci_TofCal)
+        delete fFrsSci_TofCal;
+    if (fMusli_Cal)
+        delete fMusli_Cal;
+    if (fMusli_Hit)
+        delete fMusli_Hit;
+}
+
+InitStatus R3BOnlineSpectraFrsSciVsMusli::Init()
+{
+    LOG(info) << "R3BOnlineSpectraFrsSciVsMusli::Init()";
+
+    char Name1[255];
+
+    // try to get a handle on the EventHeader. EventHeader may not be
+    // present though and hence may be null. Take care when using.
+
+    FairRootManager* mgr = FairRootManager::Instance();
+    if (NULL == mgr)
+        LOG(fatal) << "R3BOnlineSpectraFrsSciVsMusli::Init FairRootManager not found";
+
+    fEventHeader = dynamic_cast<R3BEventHeader*>(mgr->GetObject("EventHeader."));
+    if (!fEventHeader)
+        fEventHeader = dynamic_cast<R3BEventHeader*>(mgr->GetObject("R3BEventHeader"));
```

Comment:
delete these 2 lines:
if (!fEventHeader)
        fEventHeader = dynamic_cast<R3BEventHeader*>(mgr->GetObject("R3BEventHeader"));
they are not needed anymore